frontend/app/components/wp-edit/field-types/wp-edit-select-field.module.ts

```diff
@@ -58,6 +58,15 @@ export class SelectEditField extends EditField {
     } else {
       this.setValues([]);
     }
+
+    if (this.name === 'type') {
+      const services = ['$stateParams', 'v3Path', 'halRequest'];
+      const inject = service => this.$injector.get(service);
+      const [$stateParams, v3Path, halRequest] = <any[]> services.map(inject);
+      const type = parseInt($stateParams.type);
+
+      halRequest.get(v3Path.types({type})).then(type => this.value = type);
```

Comment:
This does not belong here.

It should be placed in the wp-create controller which should send the selected value as part of the request to `/api/v3/projects/:project_id/work_packages/form` so that an additional round trip is saved. 

The code here additionally prevents the user from changing the value at all as angular will reset the value on the next digest. 

Additionally, it just does not belong here from an architectural point of view.